libmscore/cmd.cpp

```diff
@@ -2464,6 +2472,50 @@ Element* Score::selectMove(const QString& cmd)
       return el;
       }
 
+//---------------------------------------------------------
+//   cmdNextPrevSystem
+//---------------------------------------------------------
+
+void Score::cmdNextPrevSystem(ChordRest* cr, bool next)
+      {
+      // To be called from within Score::move() which will provide a valid ChordRest pointer.
+      // This utilizes the already existing functionality of ScoreView::gotoMeasure()
+      // to traverse forward/backward per-system, bringing the user to the top-left most of
+      // systems (first instrument, first measure), going to last measure on last system.
+      Measure* currentMeasure       = cr->measure();
+      System*  currentSystem        = currentMeasure->system();
+      Measure* destinationMeasure   = currentSystem->firstMeasure();
+      Segment* firstSegmentOfSystem = destinationMeasure->first();
+
+      if (currentSystem) {
+            if (next) {
+                  // [Go to Next System]:
+                  // Select the first chord/rest of the next system, but if user is already
+                  // within the final system, go to its last measure. However, if user is
+                  // currently positioned within the last measure, do nothing.
+                  destinationMeasure = currentSystem->lastMeasure()->nextMeasure();
+                  if (destinationMeasure)
+                        mscore->currentScoreView()->gotoMeasure(destinationMeasure);
+                  else if (currentMeasure != lastMeasure())
+                        mscore->currentScoreView()->gotoMeasure(lastMeasure());
+                  }
+            else {
+                  // [Go to Previous System]:
+                  // If current position is equivalent to the first chord/rest of first instrument [track 0]
+                  // of the current system, go to the previous system's first measure, so long as the user isn't
+                  // in the first measure of the entire score. Otherwise, go to current system's first measure.
+                  Segment* currentSegment = cr->segment();
+                  Segment* segmentOfFirstCR = firstSegmentOfSystem->nextChordRest(cr->track(), false)->segment();
+                  if (destinationMeasure != firstMeasure())
+                        if (currentSegment == segmentOfFirstCR)
+                              if (cr->track() == 0)
```

Comment:
I think these three conditionals can be written in one line, using `&&`.

Reply:
Yes they can. Didn't want to though since it was easier for me to keep tabs this way while doing it. Does the guideline expect compaction in all cases?

Reply:
P.S. travis is giving an error but i have no idea why, it compiles fine on my Linux machine

Reply:
> Does the guideline expect compaction in all cases?

Probably no, but I still suggest using `&&` to reduce nesting level.

Reply:
I understand. I think I have a tendency when experimenting to use nests like this because I can then check debug output if I set it up so that I know what's passing what in segments, and then tend to leave it that way or maybe remove the brackets after I'm satisfied instead of compact them with logical ands.